documentscanner/src/main/java/nz/mega/documentscanner/openCV/CropDetector.kt

```diff
@@ -0,0 +1,146 @@
+package nz.mega.documentscanner.openCV
+
+import org.opencv.core.Core
+import org.opencv.core.CvType
+import org.opencv.core.Mat
+import org.opencv.core.MatOfInt
+import org.opencv.core.MatOfPoint
+import org.opencv.core.MatOfPoint2f
+import org.opencv.core.Size
+import org.opencv.imgproc.Imgproc
+import java.util.ArrayList
+import java.util.Collections
+import java.util.Comparator
+import kotlin.math.abs
+import kotlin.math.ceil
+import kotlin.math.max
+
+object CropDetector {
+
+    private const val THRESHOLD_LEVEL = 2
+    private const val AREA_LOWER_THRESHOLD = 0.2
+    private const val AREA_UPPER_THRESHOLD = 0.98
+    private const val DOWNSCALE_IMAGE_SIZE = 600.0
+
+    fun detect(src: Mat): MatOfPoint2f? {
+        // Downscale image for better performance.
+        val ratio = DOWNSCALE_IMAGE_SIZE / max(src.width(), src.height())
+        val downscaledSize = Size(src.width() * ratio, src.height() * ratio)
+        val downscaled = Mat(downscaledSize, src.type())
+        Imgproc.resize(src, downscaled, downscaledSize)
+        val rectangles = getPoints(downscaled)
+        if (rectangles.isEmpty()) {
+            return null
+        }
+        Collections.sort(rectangles, getAreaDescendingComparator())
+        val largestRectangle = rectangles[0]
+        return OpenCvUtils.scaleRectangle(largestRectangle, 1f / ratio)
+    }
+
+    private fun getPoints(src: Mat): List<MatOfPoint2f> {
```

Comment:
Add method documentation.

Reply:
Methods on this class are related with the OpenCV algorithm inherited from the forked lib. It was created by the original author who separated the logic based on his own criteria.

Reply:
Could we have at least a minimum documentation explaining that and the parameter meaning?